misskey-websocket/src/client/builder.rs

```diff
@@ -70,6 +87,30 @@ impl WebSocketClientBuilder {
         WebSocketClientBuilder::new(url.as_str())
     }
 
+    /// Sets an additional header for the connection request.
+    pub fn header<K, V>(&mut self, key: K, value: V) -> &mut Self
+    where
+        K: TryInto<HeaderName>,
+        V: TryInto<HeaderValue>,
+        K::Error: Into<http::Error>,
+        V::Error: Into<http::Error>,
+    {
+        self.inner.and_then_mut(|inner| {
+            let result = key.try_into().err_into().and_then(|key| {
+                let value = value.try_into().err_into()?;
+                Ok((key, value))
+            });
+            match result {
+                Ok((key, value)) => {
+                    inner.headers.insert(key, value);
+                    Ok(())
+                }
+                Err(e) => Err(Error::WebSocket(Arc::new(e.into()))),
```

Comment:
This is not an `Errors from underlying tungstenite library` https://docs.rs/misskey-websocket/latest/misskey_websocket/enum.Error.html#variant.WebSocket, so you should not[^1] return this as `Error::WebSocket`. Can you add `InvalidHeader` variant to `Error` and use it here?

[^1]: the use of `Error::Network` in `misskey_http::HttpClientBuilder::header` is my bad and should be fixed